lib/base-compiler.ts

```diff
@@ -900,13 +944,91 @@ export class BaseCompiler implements ICompiler {
         );
     }
 
+    getDefaultOrOverridenToolchainPath(overrides: ConfiguredOverrides): string {
+        for (const override of overrides) {
+            if (override.value) {
+                const possible = this.compiler.possibleOverrides?.find(ov => ov.name === override.name);
+                if (possible && possible.name === CompilerOverrideType.toolchain) {
+                    return override.value;
+                }
+            }
+        }
+
+        return this.toolchainPath;
+    }
+
+    getOverridenToolchainPath(overrides: ConfiguredOverrides): string | false {
+        for (const override of overrides) {
+            if (override.value) {
+                const possible = this.compiler.possibleOverrides?.find(ov => ov.name === override.name);
+                if (possible && possible.name === CompilerOverrideType.toolchain) {
+                    return override.value;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    changeOptionsBasedOnOverrides(options: string[], overrides: ConfiguredOverrides): string[] {
+        let sysrootPath: string | undefined;
+        const overriddenToolchainPath = this.getOverridenToolchainPath(overrides);
+        if (overriddenToolchainPath) {
+            sysrootPath = getSysrootByToolchainPath(overriddenToolchainPath);
+        }
+
+        for (const override of overrides) {
+            if (override.value) {
+                const possible = this.compiler.possibleOverrides?.find(ov => ov.name === override.name);
+                if (possible) {
+                    if (possible.name === CompilerOverrideType.toolchain) {
+                        if (hasToolchainArg(options)) {
+                            options = replaceToolchainArg(options, override.value);
+                        } else {
+                            for (const flag of possible.flags) {
+                                options.push(flag.replace('<value>', override.value));
```

Comment:
nitpick! maybe extract `<value>` as `const PlaceHolder = `<value>';` and then export it and we can use it in the other spots. Then when inevitably we need to change it we only have to edit one place.